src/ch/ch_monitor.c

```diff
@@ -406,7 +406,7 @@ virCHMonitorBuildNetJson(virDomainObjPtr vm, virJSONValuePtr nets, virDomainNetD
         }
     }
     if (netdef->driver.virtio.queues) {
```

Comment:
Cloud-hypervisor is checking the following:

+        if self.fds.is_some() && self.fds.as_ref().unwrap().len() * 2 != self.num_queues {
+            return Err(ValidationError::VnetQueueFdMismatch);
+        }

In the case virtio.queues is not defined in guest domain XML, I set it to default values and sent the following json to clh:

clh input json = {"cpus":{"boot_vcpus":4,"max_vcpus":4},"
console":{"mode":"Pty"},"serial":{"mode":"Null"},"memory":{"size":4294967296},"kernel":{"path":"/home/praween/libvirt/vm/hypervisor-f
w"},"disks":[{"path":"/home/praween/libvirt/vm/focal-server-cloudimg-amd64-custom.raw"},{"path":"/home/praween/libvirt/vm/cloud-confi
g-dhcp2.raw","readonly":true}],"net":[{"fds":[26],"mac":"00:11:22:39:46:11","iommu":true,"num_queues":2}]}

This handled the validation failure.

Reply:
So in ch_process.c you can see where I've changed setting the tapfdSize to be = 1 where it wasn't defined (and another change to add similar logic for ethernet). Is that reasonable do you think?

Reply:
That may be not work. Consider an configuration like below:

<interface type='network'>
        <mac address='00:11:22:39:46:11'/>
        <driver iommu='on'/>
      <source network='default' bridge='virbr0'/>
      <model type='virtio'/>
   </interface>

In this configuration cloud-hypervisor needs a TAP FD and num_queues=2 to be passed in. 

In current version, num_queues is only set when "queues" is defined in input XML.

Reply:
Ahh see, yes num_queues isn't being passed as the check is on the netdef. I switched that over to `priv->tapfdSize` and that is happy. Unfortunately the CI isn't happy because previously cloud-hypervisor would set a default ip to the host side of the tap but now that libvirt is creating things there isn't an ip assigned so I need to make some CI updates.

Reply:
LGTM!